assets/packs/data_table/main.js

```diff
@@ -4,19 +4,66 @@ import "@glideapps/glide-data-grid/dist/index.css";
 import React from "react";
 import { createRoot } from "react-dom/client";
 import { App } from "./App";
+import { createTableSkeleton } from "./Skeleton";
 
-export async function init(ctx, data) {
-  // In Firefox and Safari, during the first load (uncached), the data
-  // grid renders the default font and the font is only updated after
-  // hovering the grid. Ensuring the font is loaded helps in Firefox.
-  await ctx.importCSS(
-    "https://fonts.googleapis.com/css2?family=JetBrains+Mono:wght@400;500;600&display=swap",
-  );
-  ctx.importCSS(
-    "https://fonts.googleapis.com/css2?family=Inter:wght@400;500&display=swap",
-  );
-  ctx.importCSS("main.css");
-
+function renderApp(ctx, data) {
   const root = createRoot(ctx.root);
   root.render(<App ctx={ctx} data={data} />);
 }
+
+async function loadStyles(ctx) {
+  const cssPromises = [
+    ctx.importCSS(
+      "https://fonts.googleapis.com/css2?family=JetBrains+Mono:wght@400;500;600&display=swap",
+    ),
+    ctx.importCSS(
+      "https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600&display=swap",
+    ),
+    ctx.importCSS("main.css"),
+  ];
+
+  // Force font loading with invisible element
+  const fontPreloader = document.createElement("div");
+  fontPreloader.setAttribute("aria-hidden", "true");
+  fontPreloader.style.cssText = `
+    position: absolute;
+    visibility: hidden;
+    pointer-events: none;
+    left: -9999px;
+  `;
+  fontPreloader.innerHTML =
+    "<span style=\"font-family:'JetBrains Mono'\">Font preload</span>" +
+    "<span style=\"font-family:'Inter'\">Font preload</span>";
+  document.body.appendChild(fontPreloader);
+
+  await Promise.all(cssPromises);
+
+  const FONT_LOAD_TIMEOUT = 500;
+
+  try {
+    if (document.fonts && document.fonts.ready) {
+      await Promise.race([
+        document.fonts.ready,
+        new Promise((resolve) => setTimeout(resolve, FONT_LOAD_TIMEOUT)),
+      ]);
+    } else {
+      await new Promise((resolve) => setTimeout(resolve, FONT_LOAD_TIMEOUT));
+    }
```

Comment:
We shouldn't do the timeout.

When `document.fonts.ready` is available, we should just wait however long it takes, it may very well take a second, and since this is only the first render, it should be fine. Otherwise, we would wait 0.5s and then run into the same issues this solves.

When `document.fonts.ready` is not available, we would *always* wait the timeout, even if the fonts are actually cached, so for most renders it would slow things down unnecessarily. Looks like it's well supported anyway.